packages/estatico-json-schema/index.js

```diff
@@ -0,0 +1,166 @@
+    info: Joi.func(),
+    error: Joi.func(),
+    debug: Joi.func(),
+  }),
+});
+
+/**
+ * Default config
+ * @param {object} env - Optional environment config, e.g. { dev: true }
+ * @return {object}
+ */
+const defaults = (/* env */) => ({
+  src: null,
+  srcBase: null,
+  watch: null,
+  plugins: {
+    setup: {
+      // Which part of the input data to validate against the schema
+      // Both default data and variants will be validated
+      getData: (content /* , filePath */) => {
+        const defaultData = content.props;
+        const variants = content.variants ? Object.values(content.variants).map(v => v.props) : [];
+
+        return [defaultData].concat(variants);
+      },
+      // Where to find the schema
+      // eslint-disable-next-line arrow-body-style
+      getSchemaPath: (content /* , filePath */) => {
+        return content.meta ? content.meta.schema : null;
+      },
+    },
+    ajv: {
+      allErrors: true,
+    },
+  },
+  logger: new Logger('estatico-json-schema'),
+});
+
+/**
+ * Task function
+ * @param {object} config - Complete task config
+ * @param {object} env - Optional environment config, e.g. { dev: true }
+ * @param {object} [watcher] - Watch file events (requires `@unic/estatico-watch`)
+ * @return {object} gulp stream
+ */
+const task = (config, env = {}, watcher) => {
+  const gulp = require('gulp');
+  const plumber = require('gulp-plumber');
+  const Ajv = require('ajv');
+  const through = require('through2');
+  const fs = require('fs');
+  const $RefParser = require('json-schema-ref-parser');
+
+  const ajv = new Ajv(config.plugins.ajv);
+
+  return gulp.src(config.src, {
+    base: config.srcBase,
+    read: false,
+  })
+
+    // Prevent stream from unpiping on error
+    .pipe(plumber())
+
+    // Decide based on watcher dependency graph which files to pass through
+    .pipe(through.obj((file, enc, done) => {
+      if (watcher && watcher.matchGraph && !watcher.matchGraph(file.path, true)) {
+        return done();
+      }
+
+      return done(null, file);
+    }))
+
+    // Validate
+    .pipe(through.obj(async (file, enc, done) => {
+      const fileName = path.relative(config.srcBase, file.path);
+      const content = require(file.path); // eslint-disable-line import/no-dynamic-require
+      const validationSchemaPath = config.plugins.setup.getSchemaPath(content, file.path);
+
+      if (!validationSchemaPath) {
+        return done();
+      }
+
+      if (!fs.existsSync(validationSchemaPath)) {
+        config.logger.error({
+          fileName,
+          message: `${validationSchemaPath} not found`,
+        }, env.dev);
+
+        return done();
+      }
+
+      const validationSchema = require(validationSchemaPath); // eslint-disable-line import/no-dynamic-require,max-len
+
+      // Resolve local references
+      const resolvedValidationSchema = await $RefParser.dereference(validationSchemaPath, validationSchema, {}); // eslint-disable-line max-len
```

Comment:
@orioltf, @yannisgu: I'm using https://www.npmjs.com/package/json-schema-ref-parser to resolve relative references, this seems to be a wide-spread solution. See example in https://github.com/unic/estatico-nou/pull/68/files#diff-5f6b721e28d914ca1ddbf7f9d5d29b73